core/src/main/scala/io/finch/DecodePath.scala

```diff
@@ -19,7 +20,7 @@ object DecodePath {
     def apply(s: String): Option[A] = fn( s)
   }
 
-  implicit val decodePath: DecodePath[String] = instance(Some.apply)
+  implicit val decodePath: DecodePath[String] = instance(s => Some(QueryStringDecoder.decodeComponent(s)))
```

Comment:
I'm thinking this should be part of `ExtractPath` endpoint (and not the decode-path type-class) so we can provide the same input guarantees across all instances (i.e., `path[Int]`, `path[Boolean]`, etc).

Reply:
Do you think there is a need to decode special characters for components that are supposed to be `1`, `true`, `false` etc?

Reply:
I think so? For example, `path[Int]` should be able to parse `1%2B1` as `1`.

Reply:
Sorry, didn't get last one

Reply:
So my guess is the following should match (and extract `1`):

```scala
scala> path[Int].apply(Input.get("/%31")).awaitValueUnsafe()
res0: Option[Int] = None
```

But I might be totally missing the fact peers never send integers url-encoded?

Reply:
To my knowledge, only unsafe characters must be encoded within URL

There is the list:
http://www.ietf.org/rfc/rfc1738.txt

Reply:
Yeah, maybe also `-` (as in `/-123`)?

The rfc also mentions that peers are welcome to encode other chars as they see a fit:

```
On the other hand, characters that are not required to be encoded
   (including alphanumerics) may be encoded within the scheme-specific
   part of a URL, as long as they are not being used for a reserved
   purpose.
```

Reply:
@ImLiar Was wondering what we decided to do here. Maybe just support percent decoding for strings is a good start?

Reply:
It looks like a more fresh [standard](https://tools.ietf.org/html/rfc3986). 
From #section-2.3:
> URIs that differ in the replacement of an unreserved character with
   its corresponding percent-encoded US-ASCII octet are equivalent: they
   identify the same resource.  However, URI comparison implementations
   do not always perform normalization prior to comparison (see Section
   6).  For consistency, percent-encoded octets in the ranges of ALPHA
   (%41-%5A and %61-%7A), DIGIT (%30-%39), hyphen (%2D), period (%2E),
   underscore (%5F), or tilde (%7E) should not be created by URI
   producers and, when found in a URI, should be decoded to their
   corresponding unreserved characters by URI normalizers.

From #section-2.4
> When a URI is dereferenced, the components and subcomponents
   significant to the scheme-specific dereferencing process (if any)
   must be parsed and separated before the percent-encoded octets within
   those components can be safely decoded, as otherwise the data may be
   mistaken for component delimiters.  The only exception is for
   percent-encoded octets corresponding to characters in the unreserved
   set, which can be decoded at any time.

So suggested approach is to slice URL and then normalize each part if necessary.